commons/src/main/java/fr/free/nrw/commons/upload/GPSExtractor.java

```diff
@@ -33,21 +69,67 @@ public String getCoords() {
         }
 
         if (exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE) == null) {
-            Log.d("Image", "Picture has no GPS info");
-            return null;
-        }
-        else {
+            imageCoordsExists = false;
+            Log.d(TAG, "Picture has no GPS info");
+
+            //Check what user's preference is for automatic location detection
+            boolean gpsPrefEnabled = gpsPreferenceEnabled();
+
+            if (gpsPrefEnabled) {
+                //If pref enabled, get current location
+                Location location = locationManager.getLastKnownLocation(provider);
+
+                if (location != null) {
+                    myLocationListener.onLocationChanged(location);
```

Comment:
Hm, but only in getCoords() can we check that the picture has no coords, and thus call getLastKnownLocation() to get the location to pass to onLocationChanged(), right? If I were to call onLocationChanged(location) immediately after requestLocationUpdates(), how would I ensure that there is a location ready to be passed to it?